src/Sulu/Bundle/SearchBundle/Controller/WebsiteSearchController.php

```diff
@@ -0,0 +1,116 @@
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\SearchBundle\Controller;
+
+use Massive\Bundle\SearchBundle\Search\SearchManagerInterface;
+use Sulu\Bundle\WebsiteBundle\Resolver\ParameterResolverInterface;
+use Sulu\Component\Rest\RequestParametersTrait;
+use Sulu\Component\Webspace\Analyzer\RequestAnalyzerInterface;
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * This controller handles the search for the website.
+ */
+class WebsiteSearchController
+{
+    use RequestParametersTrait;
+
+    /**
+     * @var SearchManagerInterface
+     */
+    private $searchManager;
+
+    /**
+     * @var RequestAnalyzerInterface
+     */
+    private $requestAnalyzer;
+
+    /**
+     * @var ParameterResolverInterface
+     */
+    private $parameterResolver;
+
+    /**
+     * @var EngineInterface
+     */
+    private $engine;
+
+    public function __construct(
+        SearchManagerInterface $searchManager,
+        RequestAnalyzerInterface $requestAnalyzer,
+        ParameterResolverInterface $parameterResolver,
+        EngineInterface $engine
+    ) {
+        $this->searchManager = $searchManager;
+        $this->requestAnalyzer = $requestAnalyzer;
+        $this->parameterResolver = $parameterResolver;
+        $this->engine = $engine;
+    }
+
+    /**
+     * Returns the search results for the given query.
+     *
+     * @param Request $request
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function queryAction(Request $request)
+    {
+        $query = $this->getRequestParameter($request, 'q', true);
+
+        $locale = $this->requestAnalyzer->getCurrentLocalization()->getLocale();
+        $webspace = $this->requestAnalyzer->getWebspace();
+
+        $queryString = '';
+        if (strlen($query) < 3) {
+            $queryString .= '+("' . self::escapeDoubleQuotes($query) . '") ';
+        } else {
+            $queryValues = explode(' ', $query);
+            foreach ($queryValues as $queryValue) {
+                if (strlen($queryValue) > 2) {
+                    $queryString .= '+("' . self::escapeDoubleQuotes($queryValue) . '" OR ' .
+                        preg_replace('/([^\pL\s\d])/u', '?', $queryValue) . '* OR ' .
+                        preg_replace('/([^\pL\s\d])/u', '', $queryValue) . '~) ';
+                } else {
+                    $queryString .= '+("' . self::escapeDoubleQuotes($queryValue) . '") ';
+                }
+            }
+        }
+
+        $hits = $this->searchManager
+            ->createSearch($queryString)
+            ->locale($locale)
+            ->index('page_' . $webspace->getKey() . '_published')
+            ->execute();
+
+        return $this->engine->renderResponse(
+            $webspace->getTemplate('search'),
```

Comment:
Now it throws an exception saying that the template is not being found. So the developer at least immediately knows what's going wrong. The only thing that can be done additionally would be to use a default template in case the one given in the workspace does not exist.

The other thing we could do is to introduce a new exception, although I am not sure if it is worth it.